include/gz/math/Inertial.hh

```diff
@@ -123,6 +166,52 @@ namespace gz
         return R * this->massMatrix.Moi() * R.Transposed();
       }
 
+      /// \brief Spatial mass matrix for body B. It does not include fluid
+      /// added mass.
+      /// The matrix is arranged as follows:
+      ///
+      /// | m          0          0          0           m * Pz    -m * Py |
+      /// | 0          m          0          -m * Pz    0           m * Px |
+      /// | 0          0          m           m * Py    -m * Px    0       |
+      /// | 0          -m * Pz     m * Py    Ixx        Ixy        Ixz     |
+      /// |  m * Pz    0          -m * Px    Ixy        Iyy        Iyz     |
+      /// | -m * Py     m* Px     0          Ixz        Iyz        Izz     |
+      ///
+      /// \return The body's 6x6 inertial matrix.
+      /// \sa SpatialMatrix
+      public: Matrix6<T> BodyMatrix() const
+      {
+        Matrix6<T> result;
+
+        result.SetSubmatrix(Matrix6<T>::TOP_LEFT,
+            Matrix3<T>::Identity * this->massMatrix.Mass());
+
+        result.SetSubmatrix(Matrix6<T>::BOTTOM_RIGHT, this->massMatrix.Moi());
+
+        auto x = this->pose.Pos().X();
+        auto y = this->pose.Pos().Y();
+        auto z = this->pose.Pos().Z();
+        auto skew = Matrix3<T>(
+            0, -z, y,
+            z, 0, -x,
+            -y, x, 0) * this->massMatrix.Mass();
+        result.SetSubmatrix(Matrix6<T>::BOTTOM_LEFT, skew);
+        result.SetSubmatrix(Matrix6<T>::TOP_RIGHT, skew.Transposed());
+
+        return result;
+      }
+
+      /// \brief Spatial mass matrix, which includes the body's inertia, as well
```

Comment:
the documentation of this API should state in which frame the API is expressed, and that frame should be `F` to match the behavior of the `Moi()` function

the constructor currently says that the added mass matrix is expressed in frame `Bi`. would it be easier to require that the added mass matrix is expressed in Frame `F`?

Reply:
Updated docs in deac75cebb3a89576101407e79bb9aac853766b7.

> would it be easier to require that the added mass matrix is expressed in Frame F?

It is definitely easier for me to implement, because I'm not sure how I'd transform a matrix from the CoM frame to the link frame. But it feels to me that if we do it like that, the API will be a bit less consistent, because the moments of inertia are expressed in the CoM frame (Bi).

I'll look around for a bit and see if I find a way of transforming the added mass matrix from one frame to another.

Reply:
Looking at other references, I'm thinking it's best to leave it in the link frame. I'm also not clear how I'd transform it.

Reply:
👍